runtime/util/moduleclasses.c

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "j9.h"
+
+IDATA
+ensureJimModules(J9JavaVM* vm, JNIEnv* env) {
+	IDATA result = 0;
+	if (NULL == vm->jimModules) {
+		omrthread_monitor_enter(vm->jlmModulesInitMutex);
+		if (NULL == vm->jimModules) {
+			jclass jimModules = (*env)->FindClass(env,
+					"jdk/internal/module/Modules");
+			if (NULL == jimModules) {
+				result = -1;
+			} else {
+				jclass globalModules = (*env)->NewGlobalRef(env, jimModules);
+				if (NULL == globalModules) {
+					omrthread_monitor_exit(vm->jlmModulesInitMutex);
```

Comment:
This yields to two calls `omrthread_monitor_exit`.

Reply:
Right, this should just be and else case.

Reply:
Oops.  Missed that.